x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/SecurityTests.java

```diff
@@ -514,6 +529,81 @@ public void testLicenseUpdateFailureHandlerUpdate() throws Exception {
         }
     }
 
+    public void testSecurityHandlerIsAlwaysInstalled() throws IllegalAccessException {
```

Comment:
I am not a fan of this test. As an unit test, I think it mixes two things:
* `Security` - Ensure a non-null restWrapper is always return by `getRestHandlerWrapper`.
* `ActionModule` - Ensure only restWrapper from `Security` can be installed

Each of the above can be tested independantly in a more unit-test-like fashion. And if we want to test the real thing, we can go for internal or external cluster. 

But I am happy to let it pass if Tim is OK with it.

Reply:
Testing the ActionModule is tricky unless we change how it works. It has a hardcoded dependency on a particular classname, so it's impossible to mock it out (see https://github.com/elastic/elasticsearch/pull/73329).

We could soften that check to only check for the package name (`org.elasticsearch.xpack.security`), or to pass the name of the allowed plugin into the constructor. Either of those would then allow it to be tested in a more unit-test-like fashion. Otherwise the strict class-name dependency makes it difficult

Reply:
The check is now relaxed to be package name. Do you think it is worth to have more unit test like tests? 
Technically, even with the previous check for exact class name, it is possible to have an unit test for `ActionModule`: Since `ActionModule` is in a different module and does not depend on xpack security, it is possible to create a stub `org.elasticsearch.xpack.security.Security` class in its test package and use that for the unit test.

Reply:
I think it's testing 2 things, and I think it would be better if they were split into
1. `testSecurityPluginInstallsRestHandlerWrapperEvenIfSecurityIsDisabled` which can simply be a test that `Security.getRestHandlerWrapper()` returns non-null even when disabled.
2. `testSecurityRestHandlerWrapperCanBeInstalled()` which tests that ActionModule doesn't reject the wrapper installed by security (essentially this test).